quizz.md

```diff
@@ -0,0 +1,213 @@
+
+When called, the method should `puts`, "Would you take this rose and help out
+an old beggar, X?", where X is the person passed into the method.
+
+Demonstrate calling the method with an argument of "young prince".
+
+Write your code here:
+```
+def offerRose (person)
+  puts "Would you take this rose and help out an old beggar, " + person + "?"
+end
+
+offerRose('young_prince')
+```
+### Question 2
+
+Assume the following hash:
+
+```
+town = {
+  residents: ["Maurice", "Belle", "Gaston"],
+  castle: {
+    num_rooms: 47,
+    residents: "Robby Benson",
+    guests: []
+  }
+}
+```
+
+Using Ruby, remove Belle from the town residents, and
+add her to the list of guests in the castle.
+
+Write your code here:
+```
+town[:castle][:guests] << town[:residents].delete("Belle")
+
+# ```
+# town[:residents].delete( "Belle" )
+# +town[:castle][:guests] = "Belle"
+# other possible solution? or is this is just deleting and creating a new value in the array?
+
+### Question 3
+
+Assume you have an array of strings representing friend's names:
+
+```ruby
+friends = ["Chip Potts", "Cogsworth", "Lumière", "Mrs. Potts"]
+```
+
+Using `.each` AND string interpolation, produce output (using `puts`) like so:
+
+```
+Belle is friends with Chip Potts
+Belle is friends with Cogsworth
+Belle is friends with Lumière
+Belle is friends with Mrs. Potts
+```
+
+Write your code here:
+```ruby
+
+friends.each do |f|
+  puts "Belle is friends with " + f
+end
+
+```
+
+## SQL, Databases, and ActiveRecord (meets Aladdin)
+
+### Question 4
+
+Describe what an ERD is, and why we create them for applications. Also give an
+example what the attributes and relationships might be for the following
+entities (no need to draw an ERD):
+<!-- Maybe clarify whether they're meant to give relationships between all four entities or... --> # I'm having a hard time understanding what you mean by this...
+
+* Genie
+* Lamp
+* Person
+* Pet
+
+Your answer:
+```
+ An entity-relationship (ER) diagram is a specialized graphic that illustrates the relationships between entities in a database. ER diagrams often use symbols to represent three different types of information. Boxes are commonly used to represent entities. Diamonds are normally used to represent relationships and ovals are used to represent attributes.
+
+ A person can have many Lamps and each lamp one Genie?
+ and a person can have one pet?... I'm unsure about how to explain this.
+```
+
+### Question 5
+
+Describe what a schema is, and how we represent a one-to-many relationship in a
+SQL database. If you need an example, you can use: people and wishes
+(one-to-many).
+
+Your answer:
+```
+A schema is a collection of database objects (as far as this hour is concerned—tables) associated with one particular database username. This username is called the schema owner, or the owner of the related group of objects. You may have one or multiple schemas in a database.
+
```

Comment:
The question is asking for the tables and columns you would use to represent a one-to-many relationship.  We were hoping to see an example using primary and foreign keys.